### PR TITLE
KV 7.1 preview features

### PR DESCRIPTION
**Pull request by g2vinay**
Need to update and re-record tests.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java [hunk 1]
```diff
@@ -89,6 +89,7 @@
         Objects.requireNonNull(vaultUrl, KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.VAULT_END_POINT_REQUIRED));
         this.vaultUrl = vaultUrl.toString();
         this.service = RestProxy.create(CertificateService.class, pipeline);
+        API_VERSION = version.getVersion();
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java]: But the field is static. That means someone passing a version to one instance of a client is changing it for all clients.

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java [hunk 2]
```diff
@@ -69,7 +69,7 @@
  */
 @ServiceClient(builder = CertificateClientBuilder.class, isAsync = true, serviceInterfaces = CertificateService.class)
 public final class CertificateAsyncClient {
-    static final String API_VERSION = "7.0";
+    final String apiVersion;
```
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java]: private?

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java [hunk 3]
```diff
@@ -89,6 +89,7 @@
         Objects.requireNonNull(vaultUrl, KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.VAULT_END_POINT_REQUIRED));
         this.vaultUrl = vaultUrl.toString();
         this.service = RestProxy.create(CertificateService.class, pipeline);
+        apiVersion = version.getVersion();
```
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java]: nit: `this.apiVersion` to be consistent with other assignments above.

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java [hunk 4]
```diff
@@ -61,7 +61,7 @@
  */
 @ServiceClient(builder = KeyClientBuilder.class, isAsync = true, serviceInterfaces = KeyService.class)
 public final class KeyAsyncClient {
-    static final String API_VERSION = "7.0";
+    final String apiVersion;
```
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java]: `private` - same for all other clients.

File: sdk/keyvault/azure-security-keyvault-secrets/CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
-## 4.2.0-beta.1 (Unreleased)
-
+## 4.2.0-beta.1 (2020-03-10)
+### Added
+- Add `recoverableDays` property to `SecretProperties`.
```
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-secrets/CHANGELOG.md]: Should changelogs also call out new service version support?

**Pull request merged**